File: BI-SI1/semestral/RestAPI/src/main/kotlin/cz/cvut/fit/caltrack/data/entity/Activity.kt
package cz.cvut.fit.caltrack.data.entity

import javax.persistence.*

@Entity
data class Activity (
    @Column(nullable = false) val name: String,
    @Column(name = "calories_per_minute", nullable = false) val caloriesPerMinute: Double,
    @Column(nullable = false) val unit: String,
    @Column(name="unit_to_minutes", nullable = false) val unitToMinutes: Double,
    override val id: Int = 0
) : IEntity(id)

File: BI-SI1/semestral/Aplikace/CalTrack/Model/Activity.swift
//
//  Activity.swift
//  CalTrack
//
//  Created by Ondřej Wrzecionko on 19.04.2021.
//

import Foundation

struct Activity : Codable, Identifiable {
    let id: Int
    let name: String
    let caloriesPerMinute: Double
    let unit: String
    let unitToMinutes: Double
}

extension Activity {
    static let preview = Activity(id: 1, name: "Běh", caloriesPerMinute: 12, unit: "min", unitToMinutes: 1)
}
